blob/fileblob/fileblob_test.go

```diff
@@ -440,3 +457,38 @@ func TestListAtRoot(t *testing.T) {
 		t.Fatalf("Expecting an EOF on next item in list, got: %#v", err)
 	}
 }
+
+func TestSkipMetadata(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dir, err := ioutil.TempDir("", "fileblob*")
+	if err != nil {
+		t.Fatalf("Got error creating temp dir: %#v", err)
+	}
+	defer os.RemoveAll(dir)
+	b := &bucket{dir: dir, opts: &Options{}}
+	bucket := blob.NewBucket(driver.Bucket(b))
+
+	for _, skipMetadata := range [...]bool{false, true} {
+		b.opts.SkipMetadata = skipMetadata
+		writer, err := bucket.NewWriter(ctx, "key", &blob.WriterOptions{
+			ContentType: "text/plain",
+		})
+		if err != nil {
+			t.Fatalf("Got error creating file: %#v", err)
+		}
+		if _, err := writer.Write([]byte("hello world")); err != nil {
+			t.Fatal(err)
+		}
+		if err := writer.Close(); err != nil {
+			t.Fatal(err)
+		}
+
+		_, err = os.Stat(filepath.Join(dir, "key"+attrsExt))
+		if !skipMetadata && os.IsNotExist(err) {
```

Comment:
I think this is backwards -- it's only verifying that a metadata file *is* written in the `false` case. It is not verifying anything in the `true` case.

I think you want something like:

```
if skipMetadata != os.IsNotExist(err) {
   ...
}
```